### PR TITLE
fix: clean template content for elementor import

### LINKED ISSUE #256: Background images from Elementor templates don't get imported
### Description

When importing a TPC Page template in Elementor, the background images are not visible, and some errors appear in the console.

### Step-by-step reproduction instructions

1. Install TPC and Elementor
2. Create a page with Elementor and insert a page template from TPC https://vertis.d.pr/i/NKolFY
3. In Elementor, there are some console errors https://vertis.d.pr/i/0cXIkC and the background image is not visible
4. On the frontend, the background image is not visible https://vertis.d.pr/i/MYJVdT
5. To fix this, you just need to edit the page in Elementor and upload a new image

### Screenshots, screen recording, code snippet or Help Scout ticket

_No response_

### Environment info

_No response_

### Is the issue you are reporting a regression

No

### PR DESCRIPTION
**Pull request by Soare-Robert-Daniel**
### Summary

Some templates for Elementor have unwanted elements like `id` for the image background. Because the image background ID is present, Elementor, when doing the rendering, is trying to find the image in Media Library, and the images do not exist in Media Library because it is served via a link from our endpoints.

A simple solution to this problem is to remove the unwanted property before inserting the imported content.

### Test instructions

- With TPC and Elementor installed, import *page template* Contact 4
- All the images should be present.

https://github.com/Codeinwp/templates-patterns-collection/assets/17597852/44edeb7a-72ed-4fc0-8e57-1e5698a11394

Thank you, @mghenciu, for the help.

Closes #256

### CODE REVIEW

**Comment by Soare-Robert-Daniel** [comment]
@preda-bogdan, the clean function is duplicated in two folders, one for `assets` (to run in TPC dashboard), and the other in  `elementor`(to run import modal in the editor).

Do you think is better to make a common js script outside of those two files to have a single import? Or the redundancy is good?

**Comment by preda-bogdan** [comment]
> @preda-bogdan, the clean function is duplicated in two folders, one for `assets` (to run in TPC dashboard), and the other in `elementor`(to run import modal in the editor).
> 
> Do you think is better to make a common js script outside of those two files to have a single import? Or the redundancy is good?

I would move that outside and use it from a single place to keep the code DRY, I don't see a reason for redundancy, it will only create confusion if we later need to modify it as we would need to change it in multiple places.

Also, please check that the E2E tests are passing.

Thank you!

File: shared/utils.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * @param {any} templateContent The template content.
+ * @param {Function} cleanFunc The function to apply on each element.
+ * @return {any} The cleaned template content.
+ */
+export const cleanTemplateContent = ( templateContent, cleanFunc ) => {
+	if ( undefined === templateContent.content ) {
+		return templateContent;
+	}
+
+	console.log( templateContent );
```
Comment (preda-bogdan) [inline shared/utils.js:27]:
```suggestion
```

**Review by preda-bogdan — APPROVED** [review]
Looks good, left a small change.

File: shared/utils.js (hunk 1 above)
Comment (Soare-Robert-Daniel) [reply shared/utils.js]: Thanks. Also, I simplified the unfined checking using `?.`

**Comment by irinelenache** [comment]
@Soare-Robert-Daniel Tested and the issue persists for me for some other templates like Services 5, Gallery 5, Contact 3, About 5
You can check here:
```
 Admin area URL: https://illustriousparcel.s2-tastewp.com/wp-admin 
 Username: irinel 
 Password: RULdi-7gqVU 
```

**Comment by Soare-Robert-Daniel** [comment]
> @Soare-Robert-Daniel Tested and the issue persists for me for some other templates like Services 5, Gallery 5, Contact 3, About 5 You can check here:

I need more cleanup rules. I see that there are many ways to add an image.

Now when you import it should be alright. ⚠️ Some images might have the broken image icon, but it is solved once you refresh the page; in frontend should be no problem.

Also, update the given instance to test with the new build 👌

**Comment by irinelenache** [comment]
@Soare-Robert-Daniel Checked again and everything's fine now, thank you 👍

**Pull request merged**